cloudplatform/connectivity-destination-service/src/test/java/com/sap/cloud/sdk/cloudplatform/connectivity/ForwardAuthTokenTest.java

```diff
@@ -0,0 +1,174 @@
+package com.sap.cloud.sdk.cloudplatform.connectivity;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Consumer;
+
+import javax.annotation.Nonnull;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import com.google.common.net.HttpHeaders;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonPrimitive;
+import com.sap.cloud.environment.servicebinding.api.ServiceBinding;
+import com.sap.cloud.sdk.cloudplatform.requestheader.RequestHeaderAccessor;
+import com.sap.cloud.sdk.cloudplatform.resilience.ResilienceConfiguration;
+
+public class ForwardAuthTokenTest
+{
+    static Collection<TestCase> createTestCases()
+    {
+        return List
+            .of(
+                // forwardAuthToken = true && authType = NoAuthentication
+                new TestCase(
```

Comment:
(minor) I feel this is a bit hard to read. Maybe consider switching to a builder as done [here](https://github.com/SAP/cloud-sdk-java/blob/main/cloudplatform/connectivity-destination-service/src/test/java/com/sap/cloud/sdk/cloudplatform/connectivity/GetOrComputeSingleDestinationCommandWithoutAllDestinationsTest.java#L100)? 

e.g.
```suggestion
                forProperties(DestinationProperty.FORWARD_AUTH_TOKEN, "true")
                        .and(DestinationProperty.AUTH_TYPE, NO_AUTHENTICATION)
                        .expectTokenForwarding()
```

where `forProperties` returns a builder for `TestCase` and `expectTokenForwarding()` as well as `expectTokenIsNotForwarded()` return `TestCase`.